python/cli/data_platform_workflows_cli/update_bundle.py

```diff
@@ -66,10 +74,22 @@ def main():
     # Full list of possible series config (unsupported) can be found under "Charm series" at https://juju.is/docs/olm/bundle
     default_series = file_data.get("series")
     for app in file_data["applications"].values():
-        app["revision"] = fetch_latest_revision(
-            app["charm"], app["channel"], app.get("series", default_series)
-        )
+        channel_map, resources = fetch_charm_info_from_store(app['charm'], app['channel'])
 
+        if latest_revision := fetch_latest_revision(channel_map, app.get("series", default_series)):
+            app["revision"] = latest_revision
+        else:
+            raise ValueError(
+                f"Revision not found for {app['charm']} on {app['channel']} for Ubuntu {app.get('series', default_series)}"
+            )
+        for resource in resources:
+            if resource["type"] == "oci-image":
+                image_path, src_creds = fetch_oci_image_metadata(resource["download"]["url"])
+                app["resources"] = {
+                    resource["name"]: resource["revision"],
+                    "oci-image": image_path,
+                    "src-creds": src_creds,
```

Comment:
I would keep them close => by using the close label names. What about `oci-image` + `oci-creds` (or even split `creds` it to `oci-username` and `oci-password` ) ?

In this case the final structure will be:
```
  postgresql-k8s:
    channel: 14/edge
    charm: postgresql-k8s
    constraints: arch=amd64
    resources:
      oci-image: docker://registry.jujucharms.com/charm/kotcfrohea62xreenq1q75n1lyspke0qkurhk/postgresql-image@sha256:3abdbc00413b065fbfea8c6a3aaad8e137790ebb3e7bf5e1f42e19cbc1861926
      oci-password: MDAxOGxvY2F0aW9uIGNoYXJtc3RvcmUKMDAzMGlkZW50aWZpZXIgOTBlODUwYWU2MzllNTZmMGU2OTc5MmYyOTQzZWY4ZDYKMDA1NGNpZCBpcy1kb2NrZXItcmVwbyBjaGFybS9rb3RjZnJvaGVhNjJ4cmVlbnExcTc1bjFseXNwa2UwcWt1cmhrL3Bvc3RncmVzcWwtaW1hZ2UKMDAxM2NpZCBhbGxvdyBwdWxsCjAwMmZzaWduYXR1cmUghzwoR8THj8usK6ytvyxvGT7UbLKl/fS9K16+68B6wUEK
      oci-username: docker-registry
      postgresql-image: 164
    revision: 398
```

Reply:
> I would keep them close => by using the close label names

What do you mean by this? :-)

> What about oci-image + oci-creds (or even split creds it to oci-username and oci-password ) ?

We can split if you prefer - my initial motivation was to make it as easy as possible for users to copy/paste it into the skopeo command:

```
skopeo copy docker://<oci-image> --src-creds <src-creds>
```

Reply:
Oh, you named them to match skopeo. Sold.

> I would keep them close => by using the close label names

What do you mean by this? :-)

Forget. I meant to keep YAML elements close to each other for better YAML readability by humans:

```
      oci-image: ...
      oci-password: ...
      oci-username: ...
      postgresql-image: 164
```

VS

```
      oci-image: ...
      postgresql-image: 164
      src-creds: ...
```

But matching to skopeo kind of make sense in light of https://documentation.ubuntu.com/snap-store-proxy/en/airgap-charmhub/#export-oci-images